packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -896,8 +971,27 @@ export default function makeKernelKeeper(kernelStorage, kernelSlog) {
       // begin with `vMM.c.o+`.  In addition to deleting the c-list entry, we
       // must also delete the corresponding kernel owner entry for the object,
       // since the object will no longer be accessible.
+      const vref = k.slice(`${vatID}.c.`.length);
+      assert(vref.startsWith('o+'), vref);
       const kref = kvStore.get(k);
-      orphanKernelObject(kref, vatID);
+      // we must delete the c-list entry, but we don't need to
+      // manipulate refcounts like the way vatKeeper/deleteCListEntry
+      // does, so just delete the keys directly
+      // vatKeeper.deleteCListEntry(kref, vref);
+      kvStore.delete(`${vatID}.c.${kref}`);
+      kvStore.delete(`${vatID}.c.${vref}`);
```

Comment:
Since keys are visited in order, we _could_ consolidate the vrefs into a single batch delete with a WHERE clause like `key >= ? AND key <= ?` (but only by extending the KVStore interface, so probably not worthwhile right now).

Reply:
yeah, and the export-data entries have to be deleted at the same time, so we need to enumerate the keys anyways, at least until we come up with something fancier than just shadowing the whole kvStore into IAVL